plugins/woocommerce/src/Internal/ProductAttributesLookup/DataRegenerator.php

```diff
@@ -401,4 +432,61 @@ private function check_can_do_lookup_table_regeneration( $product_id = null ) {
 			throw new \Exception( "Can't do product attribute lookup data regeneration: product doesn't exist" );
 		}
 	}
+
+	/**
+	 * Callback to abort the regeneration process from the Status - Tools page.
+	 *
+	 * @throws \Exception The lookup table doesn't exist, or there's no regeneration process in progress to abort.
+	 */
+	private function abort_regeneration_from_tools_page() {
+		$this->verify_tool_execution_nonce();
+
+		if ( ! $this->data_store->check_lookup_table_exists() ) {
+			throw new \Exception( "Can't abort the product attribute lookup data regeneration process: lookup table doesn't exist" );
+		}
```

Comment:
The table *must* exist after this version of WooCommerce is installed or upgraded to, this is just a safety measure to help with the "transition" (actually, right now the table is not created after WooCommerce is clean-installed, this will be fixed in a separate PR). From now on the case of the table not existing will mean that the WooCommerce install is in a broken state, and thus the best thing to do is to just ignore the feature (and anyway, the above-mentioned additional PR will include the table in the list of core tables too, thus the tool we already have to create the missing tables will handle that one too).